docs/src/pages/components/autocomplete/Playground.tsx

```diff
@@ -62,7 +62,7 @@ export default function Playground() {
         {...defaultProps}
         id="controlled-demo"
         value={value}
-        onChange={(event, newValue) => {
+        onChange={(event: any, newValue: FilmOptionType | null) => {
```

Comment:
Typescript can't discriminate union type without specifying the multiple prop, it's why we need to write concrete type here, an another solution for typescript users is setting`multiple={false}` or `multiple={undefined}` so typescript can infer right typings for onChange handler, what way do you prefer?

Reply:
Seems like a bug in typescript https://github.com/microsoft/TypeScript/issues/35769

Reply:
Great to know. Should we specify the multiple prop (which will be closer to real life usage)?

Reply:
I think that we can explicitly write `multiple={undefined}` in typescript examples (with a mention in the documentation why this is needed), and when the bug is fixed, this will not be a breaking change. Or strictly require to specify `multiple` as `true` or `false`, but it seems that this is not a great DX.

Reply:
Just to be clear, it's only needed because we spread `{...defaultProps}`? What if we duplicate the props instead?

Reply:
Nope, here is a playground to demonstrate the problem (an error in the end of the file)
https://www.typescriptlang.org/play/#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQdwAdjFlAXlnAKqpYCCAVxgRcJMABssrALKCJMYJKwAFYmFQAeACoA+OAG86cE3AhhFEJqgBccbQG0AuvVNwQ8xcrswogrK6mVgDCABbITADmWAD8dgAUWABuWCx22HgwAHRhEdEAoiksmgYAvroANHBJyBL+do5OAJRwALz6SRDAACb0pQzMrOycPHxCImLgUqwAysyRUmrmWnqGxkEWwFa29s6BJh4KSlJxcBwSfPtmTLlRsQnJqTDpVNm3BUUwJeVVNXVYDTgAB84Ex5BIWu1ql1enR+nQYABPMBcXgCYSicTTVTqFb6VrrEwgtHjTFTaRYOZRRa4nS6QnA0boiZYilyI7KJYaOn0OgEQRMfBbJhwQRjDGTZSsOnxBlgXF2EkS1msLl4ip0FpGNyUGCCKAi3z+Pq8sXMslSrDxbWmQ5eKQ+PxYDVucyWax2BxkZBkKpkABGZCcLqCN3Cdwen1+tX8kP0BjgpTgAHpk9UY1xgOhUL55s4qgRmFg4U0GGbSZLsdaGXbjgCzrU+CGTG7hTsvT6-YHgwyQuHopGntH-nHDImU2m-v44Fm4DmoPMzFBQeCC0WS2XxSzyaxq25a95RUxulhC0wsN1m2ZNttPd7feRu1e+3l64ko+mR214+PU5-p7O86LtAK4SBIa7nhudDlsqO5WjaLY3h6cAdg+AZBs+YavoOLDDrG35jkmf6PCwKDHv+mboBEiJVP6yDdBuQA

Reply:
OK thanks, in this can, I think that it would be better not to change the current tradeoff. The usage of the TypeScript demo is at 10% (and growing). It would look weird to see undefined. What do you think?

Reply:
If a typescript developer has not defined the `multiple` props as `false` or `undefined` then all other dependent props will also receive invalid typings, not only the `onChange` prop.

Reply:
If a typescript developer has not defined the multiple prop as false or undefined than all other dependent props will also receive invalid typings, not only the onChange prop.

Reply:
What about we document it in the `## Limitations` section of the page for TypeScript users, with a link to the related TypeScript issue?

Reply:
Sounds good.

Reply:
This would be my proposal, I believe we already get significant improvements with your changes. This is a roadblock that will require [more effort](https://github.com/microsoft/TypeScript/issues/35769).